src/MongoDB.Driver.Core/Core/Operations/QueryHelper.cs

```diff
@@ -52,21 +52,28 @@ public static int CalculateFirstBatchSize(int? limit, int? batchSize)
             return firstBatchSize;
         }
 
-        public static BsonDocument CreateReadPreferenceDocument(ServerType serverType, ReadPreference readPreference)
+        public static BsonDocument CreateReadPreferenceDocument(ServerType serverType, ReadPreference readPreference, out bool slaveOk)
```

Comment:
When Dmitry first reviewed this pull request he asked why we didn't call this overload from everywhere. At first I thought he was right and I was going to change the driver to always call this method, but then I realized that this overload is only appropriate when the caller is willing and able to set the slaveOk bit in the OP_QUERY message, and not all callers are.

By adding the `out bool slaveOk` parameter we achieve two things:

1. signal that the caller has to set the slaveOk bit
2. centralize where we determine what value the slaveOk bit should set to